### PR TITLE
Promtool: add http config support to query commands

### PR DESCRIPTION
**Pull request by FUSAKLA**
Hi, 

as a followup to the previous PR from Alertmanager https://github.com/prometheus/alertmanager/pull/2764 and stale PR for promtool https://github.com/prometheus/prometheus/pull/8082

This PR adds support for [HTTP config](https://prometheus.io/docs/alerting/latest/configuration/#http_config) using the promconfig common package same as amtool does for query subcommands.

Works just fine, tested with base auth and bearer token.

@roidelapluie would you mind taking a look since you were actibe in both above linked PRs? :pray:

### CODE REVIEW

**Comment by FUSAKLA** [comment]
@dgl would you find a second to review please?:pray:

**Comment by roidelapluie** [comment]
Can we move this code to Prometheus/common?

**Comment by FUSAKLA** [comment]
> Can we move this code to Prometheus/common?

Which part of the code do you exactly mean to move to the common? The `amconfig.LoadHTTPConfigFile`?

**Comment by roidelapluie** [comment]
yes please

**Comment by FUSAKLA** [comment]
Actually it appears the function is already there, just in the test files and not used, nor by Alertmanager or Prometheus itself.
Alertmanager has one additional line in that function, not sure hot to deal with this difference. 

PR here https://github.com/prometheus/common/pull/415

**Comment by FUSAKLA** [comment]
Switched to prometheus/common as agreed after https://github.com/prometheus/common/pull/415

**Comment by FUSAKLA** [comment]
Rebased, @roidelapluie friendly bump if you would find awhile :pray:

**Comment by FUSAKLA** [comment]
Rebased again

**Comment by FUSAKLA** [comment]
Rebased, fuzzing seems to be flaky?

**Comment by bboreham** [comment]
Fuzzing issue #11810

**Comment by bboreham** [comment]
Would you consider including `tsdb create-blocks-from rules`, which also performs queries ?
(no pressure, just asking)

**Comment by Pokom** [comment]
I have a basic version of this working here: https://github.com/Pokom/prometheus/commit/94094bf176f612276f1cc62dfb51d785df8ba7f4. For the first pass I opted to get the code working where the flag isn't shared between the two subcommands. 

If FUSAKLA isn't keen to implement the feature in this PR, I'd be more then happy to follow up with a PR to implement it.

**Comment by FUSAKLA** [comment]
Hi, sorry for delay. Sure, I'll add the support event for the `tsdb create-blocks-from rules`.

**Comment by FUSAKLA** [comment]
@bboreham PTAL 

@roidelapluie friendly bump once again :pray:

**Comment by roidelapluie** [comment]
can you run go mod tidy?

**Comment by FUSAKLA** [comment]
I ran it multiple times locally (go 1.19) and in container (official 1.18) with no changes :thinking:
Running whole `make` passes

**Comment by FUSAKLA** [comment]
I'm not able to reproduce anyhow locally the issues CI bumps into. 
My changed does not change any dependencies, so I do not see reason for it to change to cause changes in `go.mod` or `go.sum` :/

**Comment by bboreham** [comment]
I think there is some general issue with CI - I've seen these error messages on other PRs.
Didn't yet figure out what the issue is.

**Comment by FUSAKLA** [comment]
I suspected bump in the golang-builder on Go 1.20, but that seems also not to be an issue… :thinking: 
Might be worth trying to run on main branch to see if it fails only on branches, seeing other PRs with the same issues as Bryan sated

**Comment by FUSAKLA** [comment]
:tada: rebased on master and CI passes.. 

@roidelapluie Would you take a look :pray:

**Review by bboreham — APPROVED** [review]
lgtm

File: cmd/promtool/main.go [hunk 1]
```diff
@@ -303,7 +328,7 @@ func main() {
 		os.Exit(backfillOpenMetrics(*importFilePath, *importDBPath, *importHumanReadable, *importQuiet, *maxBlockDuration))
 
 	case importRulesCmd.FullCommand():
-		os.Exit(checkErr(importRules(*importRulesURL, *importRulesStart, *importRulesEnd, *importRulesOutputDir, *importRulesEvalInterval, *maxBlockDuration, *importRulesFiles...)))
+		os.Exit(checkErr(importRules(serverURL, *importRulesStart, *importRulesEnd, *importRulesOutputDir, *importRulesEvalInterval, *maxBlockDuration, *importRulesFiles...)))
```
Comment (roidelapluie) [inline cmd/promtool/main.go:331]: Can this one also take the roundtripper?

**Comment by roidelapluie** [comment]
> Hi, sorry for delay. Sure, I'll add the support event for the tsdb create-blocks-from rules.

Does not seem to be done

File: cmd/promtool/main.go (hunk 1 above)
Comment (FUSAKLA) [reply cmd/promtool/main.go]: :flushed: sorry about that, not sure how that slipped thought… 

Thanks for noticing, should be fixed  now

Comment (FUSAKLA) [reply cmd/promtool/main.go]: https://github.com/prometheus/prometheus/pull/11487/commits/f1de2cec3d34a94d594ae727e86cd18ef44d8f39

**Comment by FUSAKLA** [comment]
Should be fixed now :crossed_fingers:

**Comment by roidelapluie** [comment]
Thanks!

**Comment by FUSAKLA** [comment]
Thank you! :tada:

**Pull request merged**